python/lsst/ctrl/bps/wms/htcondor/lssthtc.py

```diff
@@ -375,43 +433,65 @@ def htc_submit_dag(htc_dag, submit_options=None):
         sub.queue(txn, ad_results=jobs_ads)
 
     # Submit.queue() above will raise RuntimeError if submission fails, so
-    # 'job_ad' should contain the ad at this point.
+    # 'jobs_ads' should contain the ad at this point.
     dag_ad = jobs_ads[0]
 
-    htc_dag.run_id = f"{dag_ad['ClusterId']}.{dag_ad['ProcId']}"
-
-    # Store additional pieces of information (e.g. 'GlobalJobId') which cannot
-    # be retrieved later from the files produced by HTCondor. Sadly,
-    # the ClassAd from Submit.queue() (see above) does not have 'GlobalJobId'.
-    # So we need to run a fully fledged query to get it anyway.
+    # Sadly, the ClassAd from Submit.queue() (see above) does not have
+    # 'GlobalJobId' so we need to run a regular query to get it anyway.
     schedd_name = schedd_ad["Name"]
-    dag_info = condor_q(constraint=f"ClusterId == {int(float(htc_dag.run_id))}",
-                        schedds={schedd_name: schedd})
-    if dag_info:
-        write_dag_info(f"{htc_dag.name}.info.json", dag_info)
-    else:
-        _LOG.debug("DAGMan job with id '%s' not found", htc_dag.run_id)
+    schedd_dag_info = condor_q(constraint=f"ClusterId == {dag_ad['ClusterId']}",
+                               schedds={schedd_name: schedd})
+    return schedd_dag_info
 
 
-def _htc_submit_dag_old(dag_filename, submit_options=None):
-    """Call condor_submit_dag on given dag description file.
-    (Use until using condor version with htcondor.Submit.from_dag)
+def htc_create_submit_from_dag(dag_filename, submit_options=None):
+    """Create a DAGMan job submit description.
 
     Parameters
     ----------
     dag_filename : `str`
         Name of file containing HTCondor DAG commands.
-    submit_options : `dict`, optional
+    submit_options : `dict` [`str`, Any], optional
         Contains extra options for command line (Value of None means flag).
 
     Returns
     -------
     sub : `htcondor.Submit`
-        htcondor.Submit object created for submitting the DAG
+        An object representing a job submit description.
+
+    Notes
+    -----
+    Use with HTCondor versions which does support htcondor.Submit.from_dag,
+    i.e., 8.9.3 or newer.
     """
+    return htcondor.Submit.from_dag(dag_filename, submit_options)
+
+
+def htc_create_submit_from_cmd(dag_filename, submit_options=None):
+    """Create a DAGMan job submit description.
+
+    Create a DAGMan job submit description by calling ``condor_submit_dag``
+    on given DAG description file.
 
+    Parameters
+    ----------
+    dag_filename : `str`
+        Name of file containing HTCondor DAG commands.
+    submit_options : `dict`, optional
```

Comment:
Because you fixed the previous one, I noticed this one also needs the  [`str`, Any].